docs/source/model_design/programming_vivarium/artifact_building/index.rst

```diff
@@ -29,12 +29,156 @@ Artifact Building
 
 .. contents::
    :local:
-   :depth: 1
+   :depth: 2
 
-.. todo::
+A data artifact is a bundle of input data associated with a particular
+model. It is typically stored as an :code:`hdf` file on disk with very
+particular formatting. This file is then used by the vivarium simulations
+to fill in all the relevant parameter data.
 
-  Add to this page. Include how-to type information on building the artifact, targeted to a research team audience. Include links to the engineering documentation as well. 
+There are also artifact building instructions in the `vivarium tutorial docs <https://vivarium.readthedocs.io/en/latest/tutorials/artifact.html>`_. However,
+these are largely written for a software developer, so here we endeavor to
+create documentation from a research team perspective.
 
-  Review and integrate this `engineering page <https://vivarium.readthedocs.io/en/latest/tutorials/artifact.html>`_. 
+Understanding Data Keys
+-----------------------
 
+A data key in a simulation is a key word that defines a set of
+data to be saved in the artifact. For example, :code:`population.location`,
+:code:`cause.measles.prevalence`, or :code:`risk_factor.child_wasting.exposure`.
 
+You should be used to seeing data keys like this in the artifact, since
+research team members often use these in V&V. In fact, this is how the data
+is called to create the artifact, and called from the artifact by research to
+test the model.
+
+These data keys show up in two places:
+
+#. The :code:`data_keys.py` file
```

Comment:
I would include the containing directory here, so constants/data_keys.py and data/loader.py